### PR TITLE
Added keyboard KPRepublic BM60V2EC

### PR DESCRIPTION
**Pull request by LuisGit9**
Added support for the KPRepublic BM60 with Encoder, Revision 2 based on the code for the BM60HSRGB Rev 2

## Description
Copied the kprepublic/bm60hsrgb directory and added support for the encoder as a button and rotation in both direction, assigning it as a mouse wheel and volumen control

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x ] Keyboard (addition or update)
- [ x ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ x ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kprepublic/bm60hsrgb_ec/rev2/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        if (clockwise) {
+          tap_code(KC_WH_D);
+        } else {
+          tap_code(KC_WH_U);
+        }
+      }
+  }
+
+  return false;
+}
```
Comment (drashna) [inline keyboards/kprepublic/bm60hsrgb_ec/rev2/keymaps/default/keymap.c:62]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_ec/rev2/rev2.c [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+// clang-format off
+const rgb_matrix_driver_t rgb_matrix_driver = {
+    .init          = rgb_matrix_driver_init,
+    .flush         = rgb_matrix_driver_flush,
+    .set_color     = rgb_matrix_driver_set_color,
+    .set_color_all = rgb_matrix_driver_set_color_all,
+};
+// clang-format on
+
+#endif /* RGB_MATRIX_ENABLE */
```
Comment (drashna) [inline keyboards/kprepublic/bm60hsrgb_ec/rev2/rev2.c:200]:
```suggestion
#endif /* RGB_MATRIX_ENABLE */

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) {
        return false;
    }
    if (layer_state_is(1)) {
	      if (clockwise) {
	          tap_code(KC_VOLU);
	      } else {
	          tap_code(KC_VOLD);
	      }
	} else {
        // ctrl
        if (get_mods() & MOD_BIT(KC_LCTRL)) {
            if (clockwise) {
                tap_code(KC_WH_R);
            } else {
                tap_code(KC_WH_L);
            }
        } else {
            if (clockwise) {
                tap_code(KC_WH_D);
            } else {
                tap_code(KC_WH_U);
            }
        }
    }
 
    return false;
}
#endif
```

File: keyboards/kprepublic/bm60hsrgb_ec/rev2/rules.mk [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+NO_USB_STARTUP_CHECK = no
```
Comment (drashna) [inline keyboards/kprepublic/bm60hsrgb_ec/rev2/rules.mk:19]:
```suggestion

```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kprepublic/bm60hsrgb_ec/readme.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# BM60HSRGB_EC
+
+A 60% hotswap inswitch RGB keyboard with rotary encoder from KP Republic. 
+
+* Keyboard Maintainer: [luisgit9](https://github.com/luisgit9)
+* Hardware Supported: BM60HSRGB_EC/rev2
+* Hardware Availability: [KP Republic](https://kprepublic.com/products/bm60ec-bm60-ec-rgb-60-gh60-hot-swappable-custom-mechanical-keyboard-pcb-programmed-qmk-via-rgb-switch-type-c-rotary-knob)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/kprepublic/bm60hsrgb_ec/readme.md:9]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/kprepublic/bm60hsrgb_ec/rev2/rev2.c (hunk 2 above)
Comment (LuisGit9) [reply keyboards/kprepublic/bm60hsrgb_ec/rev2/rev2.c:200]: This is the matching #endif for line 19 (#ifdef RGB_MATRIX_ENABLE). Should that be removed as well? Please provide guidance.

File: keyboards/kprepublic/bm60hsrgb_ec/rev2/keymaps/default/keymap.c (hunk 1 above)
Comment (LuisGit9) [reply keyboards/kprepublic/bm60hsrgb_ec/rev2/keymaps/default/keymap.c:62]: Would this change disable the encoder's functionality? Is it best practice to leave that functionality undefined and in the default keymap so the user defines that functionality? Please advise.

File: keyboards/kprepublic/bm60hsrgb_ec/readme.md (hunk 4 above)
Comment (LuisGit9) [reply keyboards/kprepublic/bm60hsrgb_ec/readme.md:9]: Added instructions to the keyboard readme.md, as well as on the rev2 readme.md.

**Review by LuisGit9 — COMMENTED** [review]
Some questions about two of the suggested changes.

File: keyboards/kprepublic/bm60hsrgb_ec/rev2/rev2.c (hunk 2 above)
Comment (drashna) [reply keyboards/kprepublic/bm60hsrgb_ec/rev2/rev2.c:200]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.


As for the endif, the "removal" is just how git shows the diff.

**Review by LuisGit9 — COMMENTED** [review]
Changes applied.

File: keyboards/kprepublic/bm60hsrgb_ec/rev2/rev2.c (hunk 2 above)
Comment (LuisGit9) [reply keyboards/kprepublic/bm60hsrgb_ec/rev2/rev2.c:200]: Committed this twice by mistake, inserted code twice. Fixed.

File: keyboards/kprepublic/bm60hsrgb_ec/rev2/config.h [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+#define RGBLED_NUM          6
+#define RGB_DI_PIN          F0
+#define RGBLIGHT_ANIMATIONS
+#define RGBLIGHT_DEFAULT_MODE RGBLIGHT_MODE_RAINBOW_MOOD
+#define RGBLIGHT_HUE_STEP   8
+#define RGBLIGHT_SAT_STEP   8
+#define RGBLIGHT_VAL_STEP   8
+
+#define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
+#define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
```
Comment (drashna) [inline keyboards/kprepublic/bm60hsrgb_ec/rev2/config.h:63]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by LuisGit9 — COMMENTED** [review]
Accepted change.

File: keyboards/kprepublic/bm60hsrgb_ec/rev2/config.h [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+#define DIODE_DIRECTION ROW2COL
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 3
+
+//rgb light setting
+#define RGBLIGHT_LIMIT_VAL 150
+#define RGBLED_NUM          6
+#define RGB_DI_PIN          F0
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/kprepublic/bm60hsrgb_ec/rev2/config.h:56]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by LuisGit9 — COMMENTED** [review]
Change accepted.

**Comment by Didel** [comment]
It looks like this PR is similar to (the older) https://github.com/qmk/qmk_firmware/pull/15104 . As #15104 adds both v1 and v2 support, I looks like it will add the same functionality as this PR with support for more revisions of the board...

**Review by drashna — APPROVED** [review]

File: keyboards/kprepublic/bm60hsrgb_ec/rev2/info.json [hunk 7]
```diff
@@ -0,0 +1,79 @@
+{
+    "keyboard_name": "BM60V2EC",
+    "url": "https://kprepublic.com/products/bm60ec-bm60-ec-rgb-60-gh60-hot-swappable-custom-mechanical-keyboard-pcb-programmed-qmk-via-rgb-switch-type-c-rotary-knob",
```
Comment (keyboard-magpie) [inline keyboards/kprepublic/bm60hsrgb_ec/rev2/info.json:3]:
```suggestion
     "manufacturer": "kprepublic",
     "url": "https://kprepublic.com/products/bm60ec-bm60-ec-rgb-60-gh60-hot-swappable-custom-mechanical-keyboard-pcb-programmed-qmk-via-rgb-switch-type-c-rotary-knob",
```

**Pull request closed without merging**